src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureBillingProjectWizard.ts

```diff
@@ -0,0 +1,160 @@
+  }, [billingProjectName, existingProjectNames])
+
+  const stepFinished = (step: number, finished: boolean) => {
+    if (finished && activeStep === step) { // the user completed the active step
+      setActiveStep(step + 1)
+    } else if (!finished && activeStep > step) { // the user went back
+      setActiveStep(step)
+    } // the user is entering fields for later steps - don't change active step
+  }
+
+  const onManagedAppSelected = managedApp => {
+    stepFinished(1, !!managedApp)
+    setManagedApp(managedApp)
+  }
+
+  const step1HasNoErrors = !!subscriptionId && !!managedApp
+  const step2HasNoErrors = addUsersOrOwners === false ||
+    (addUsersOrOwners === true && !ownerEmails.hasError && !userEmails.hasError &&
+        (ownerEmails.emails.trim().length > 0 || userEmails.emails.trim().length > 0))
+
+  return h(Fragment, [h(StepWizard, {
+    title: 'Link an Azure Subscription to Terra',
+    intro: `The linked subscription is required to cover all Azure data storage, compute and egress costs incurred in a Terra workspace.
+        Cloud costs are billed directly from Azure and passed through Terra billing projects with no markup.`
+  }, [
+    h(AzureSubscriptionStep, {
+      isActive: activeStep === 1,
+      subscriptionId,
+      onSubscriptionIdChanged: subscriptionId => {
+        stepFinished(1, false)
+        setSubscriptionId(subscriptionId)
+        onManagedAppSelected(undefined)
+      },
+      managedApp,
+      onManagedAppSelected,
+      isBusy,
+      setIsBusy
+    }),
+    h(AddUsersStep, {
+      userEmails: userEmails.emails,
+      ownerEmails: ownerEmails.emails,
+      addUsersOrOwners,
+      onAddUsersOrOwners: addUsersOrOwners => {
+        stepFinished(2, !addUsersOrOwners)
+        setAddUsersOrOwners(addUsersOrOwners)
+      },
+      onSetUserEmails: (emails, hasError) => {
+        stepFinished(2, false)
+        setUserEmails({ emails, hasError })
+      },
+      onSetOwnerEmails: (emails, hasError) => {
+        stepFinished(2, false)
+        setOwnerEmails({ emails, hasError })
+      },
+      onOwnersOrUsersInputFocused: () => {
+        stepFinished(2, false)
+      },
+      isActive: activeStep === 2
+    }),
+    h(CreateNamedProjectStep, {
+      billingProjectName: billingProjectName !== undefined ? billingProjectName : '',
+      onBillingProjectNameChanged: billingProjectName => {
+        setBillingProjectName(billingProjectName)
+        Ajax().Metrics.captureEvent(Events.billingAzureCreationProjectNameEntered)
+      },
+      onBillingProjectInputFocused: () => {
+        if (step1HasNoErrors && step2HasNoErrors) {
+          stepFinished(2, true)
+        }
+      },
+      createBillingProject,
+      projectNameErrors,
+      isActive: activeStep === 3,
+      createReady: step1HasNoErrors && step2HasNoErrors && !!billingProjectName && !projectNameErrors && !isBusy
+    })
+  ]),
+  isBusy && spinnerOverlay]) // TODO: overlay problem: not filling the whole screen.
```

Comment:
I'm pretty sure this was a pre-existing issue, but it's very noticeable with this long view. It would be great to fix it, but probably as a follow-up ticket.